### PR TITLE
Support rename of functions across class hiearchies and protocols

### PR DESCRIPTION
**Pull request by ahoppen**
rdar://123536502

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

File: Sources/SourceKitLSP/Rename.swift [hunk 1]
```diff
@@ -669,26 +700,60 @@ extension SourceKitServer {
...
+      let serverType = LanguageServerType(symbolProvider: index.symbolProvider(for: url.path))
+      languageServerTypesCache[url] = serverType
+      return serverType
+    }
+
+    let usrsToRename = overridingAndOverriddenUsrs(of: usr, index: index)
+    for usrToRename in usrsToRename {
+      for occurrence in index.occurrences(ofUSR: usrToRename, roles: renameRoles) {
+        let url = URL(fileURLWithPath: occurrence.location.path)
+        if changes[DocumentURI(url)] != nil {
```
Comment (bnbarham) [inline Sources/SourceKitLSP/Rename.swift:718]: Can this be a guard instead, I completely missed this if originally 😅

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**